dfget/core/downloader/p2p_downloader/client_writer.go

```diff
@@ -34,6 +34,13 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+// RunWaiter will be used in p2p downloader
+//
+type RunWaiter interface {
```

Comment:
I think that the interface name is not very clear.
```
1. pre-steps: initialize
2. writing: run and wait to write piece data to target
3. post-steps: finish a task
```
In addition, it's better to provide a method to put data, thus the concrete implementation of `RunWaiter` can implements different queue because the `pkg/queue` is not very suitable for streaming writer.